src/java/org/apache/cassandra/service/reads/ReadCallback.java

```diff
@@ -94,9 +191,9 @@ void addTombstoneAbort(InetAddressAndPort from, int tombstones)
             = AtomicIntegerFieldUpdater.newUpdater(ReadCallback.class, "failures");
     private volatile int failures = 0;
     private final Map<InetAddressAndPort, RequestFailureReason> failureReasonByEndpoint;
-    private volatile WarningCounter warningCounter;
-    private static final AtomicReferenceFieldUpdater<ReadCallback, ReadCallback.WarningCounter> warningsUpdater
-        = AtomicReferenceFieldUpdater.newUpdater(ReadCallback.class, ReadCallback.WarningCounter.class, "warningCounter");
+    private volatile WarningContext warningContext;
```

Comment:
Question: The idea here is that even with the guardrail enabled, most queries won't warn or abort, so be lazy?

Reply:
yes